llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -730,10 +731,13 @@ void addInstrRequirements(const MachineInstr &MI,
     break;
   }
   case SPIRV::OpTypePointer: {
+    if (!ST.isOpenCLEnv())
+      break;
     auto SC = MI.getOperand(1).getImm();
     Reqs.getAndAddRequirements(SPIRV::OperandCategory::StorageClassOperand, SC,
                                ST);
-    // If it's a type of pointer to float16, add Float16Buffer capability.
+    // If it's a type of pointer to float16 targeting OpenCL, add Float16Buffer
+    // capability.
```

Comment:
Hm... it looks like you don't want to execute these lines for Vulkan:
```
auto SC = MI.getOperand(1).getImm();
Reqs.getAndAddRequirements(SPIRV::OperandCategory::StorageClassOperand, SC, ST);
```
however, in the previous patch, you were going to disable only the code after the comment.

Reply:
Oops, good catch, fixed!